### ISSUE TITLE
Crash when stroking a circle

### ISSUE THREAD

**Issue by j-f1** [TokamakUI/Tokamak#322]
The following code:

```swift
Circle().stroke(Color.gray)  
```

produces this error. Explicitly specifying `.colorScheme(.light)` does not help.

<details>

```
Fatal error: _ColorSchemeKey must have a renderer-provided default value: file TokamakCore/ColorScheme.swift, line 22
Swift._assertionFailure(_: Swift.StaticString, _: Swift.String, file: Swift.StaticString, line: Swift.UInt, flags: Swift.UInt32) -> Swift.Never
static TokamakCore._ColorSchemeKey.defaultValue.getter : TokamakCore.ColorScheme
protocol witness for static TokamakCore.EnvironmentKey.defaultValue.getter : A.Value in conformance TokamakCore._ColorSchemeKey : TokamakCore.EnvironmentKey in TokamakCore
TokamakCore.EnvironmentValues.subscript.getter : <A where A: TokamakCore.EnvironmentKey>(A.Type) -> A.Value
TokamakCore.EnvironmentValues.colorScheme.getter : TokamakCore.ColorScheme
TokamakCore._SystemColorBox.resolve(in: TokamakCore.EnvironmentValues) -> TokamakCore.AnyColorBox._RGBA
TokamakCore._ColorProxy.resolve(in: TokamakCore.EnvironmentValues) -> TokamakCore.AnyColorBox._RGBA
(extension in TokamakStaticHTML):TokamakCore.Color.cssValue(TokamakCore.EnvironmentValues) -> Swift.String
(extension in TokamakStaticHTML):TokamakCore._StrokedShape.attributes(TokamakCore.ShapeStyle) -> [TokamakStaticHTML.HTMLAttribute : Swift.String]
protocol witness for TokamakStaticHTML.ShapeAttributes.attributes(TokamakCore.ShapeStyle) -> [TokamakStaticHTML.HTMLAttribute : Swift.String] in conformance TokamakCore._StrokedShape<A> : TokamakStaticHTML.ShapeAttributes in TokamakStaticHTML
(extension in TokamakStaticHTML):TokamakCore._ShapeView.deferredBody.getter : TokamakCore.AnyView
protocol witness for TokamakCore.ViewDeferredToRenderer.deferredBody.getter : TokamakCore.AnyView in conformance TokamakCore._ShapeView<A, B> : TokamakCore.ViewDeferredToRenderer in TokamakStaticHTML
closure #1 (Any) -> TokamakCore.AnyView in TokamakCore.AnyView.init<A where A: TokamakCore.View>(A) -> TokamakCore.AnyView
TokamakCore.StackReconciler.render<A>(compositeElement: TokamakCore.MountedCompositeElement<A>, body: Swift.ReferenceWritableKeyPath<TokamakCore.MountedCompositeElement<A>, Any>, result: Swift.KeyPath<TokamakCore.MountedCompositeElement<A>, (Any) -> A1>) -> A1
TokamakCore.StackReconciler.render(compositeView: TokamakCore.MountedCompositeView<A>) -> TokamakCore.AnyView
TokamakCore.MountedCompositeView.mount(before: A.TargetType?, on: TokamakCore.MountedElement<A>?, with: TokamakCore.StackReconciler<A>) -> ()
TokamakCore.MountedScene.mount(before: A.TargetType?, on: TokamakCore.MountedElement<A>?, with: TokamakCore.StackReconciler<A>) -> ()
TokamakCore.MountedApp.mount(before: A.TargetType?, on: TokamakCore.MountedElement<A>?, with: TokamakCore.StackReconciler<A>) -> ()
TokamakCore.StackReconciler.(performInitialMount in _FCAA3CDB23B668BF64A31B25B678943F)() -> ()
TokamakCore.StackReconciler.init<A where A1: TokamakCore.App>(app: A1, target: A.TargetType, environment: TokamakCore.EnvironmentValues, renderer: A, scheduler: (() -> ()) -> ()) -> TokamakCore.StackReconciler<A>
TokamakCore.StackReconciler.__allocating_init<A where A1: TokamakCore.App>(app: A1, target: A.TargetType, environment: TokamakCore.EnvironmentValues, renderer: A, scheduler: (() -> ()) -> ()) -> TokamakCore.StackReconciler<A>
TokamakDOM.DOMRenderer.init<A where A: TokamakCore.App>(A, JavaScriptKit.JSObject, TokamakCore.EnvironmentValues?) -> TokamakDOM.DOMRenderer
TokamakDOM.DOMRenderer.__allocating_init<A where A: TokamakCore.App>(A, JavaScriptKit.JSObject, TokamakCore.EnvironmentValues?) -> TokamakDOM.DOMRenderer
static (extension in TokamakDOM):TokamakCore.App._launch(A, TokamakCore.EnvironmentValues, JavaScriptKit.JSObject) -> ()
static (extension in TokamakDOM):TokamakCore.App._launch(A, TokamakCore.EnvironmentValues) -> ()
protocol witness for static TokamakCore.App._launch(A, TokamakCore.EnvironmentValues) -> () in conformance swiftui_tut.TokamakApp : TokamakCore.App in swiftui_tut
static (extension in TokamakCore):TokamakCore.App.main() -> ()
main
__main_argc_argv
__main_void
__original_main
_start
```

**Comment by mortenbekditlevsen** [comment]
The issue appears to have something to do with the environment not being correctly propagated to the `_ShapeView` when the stroke is applied.
I experienced the same issue in TokamakGTK, but I am not familiar enough with the environment injection to figure out why it's not being injected.

**Comment by mortenbekditlevsen** [comment]
A bit more info that may or may not be of use for debugging:

Depending on how types are wrapped on the call site, the environment is either injected correctly or not.
I made the tests with TokamakGTK, but it should be the same with DOM/HTML.

The lines marked with '// crashes' are those that don't get the environment injected.

```
struct MyShape: Shape {
  func path(in rect: TokamakGTK.CGRect) -> Path {
    var path = Path()
    path.addEllipse(
      in: CGRect(origin: .zero, size: .init(width: 100, height: 150)),
      transform: TokamakCore.CGAffineTransform(rotationAngle: 0.5)
    )
    return path.applying(TokamakCore.CGAffineTransform(rotationAngle: -0.1))
  }
}

struct MyStrokedShape: View {
  var body: some View {
    MyShape()
      .stroke(
        Color.red,
        style: StrokeStyle(lineWidth: 4, lineCap: .round, lineJoin: .round, dash: [5, 5, 1])
      )
  }
}

struct MyStrokedAndFramedShape: View {
  var body: some View {
    MyShape()
      .stroke(Color.red)
      .frame(width: 50, height: 200)
  }
}

struct TokamakGTKDemo: App {
  @State var text: String = "Test"
  var body: some Scene {
    WindowGroup("Test Scene") {
      MyShape()
        .frame(width: 50, height: 200) // works

      MyShape()
        .stroke(Color.red)
        .frame(width: 50, height: 200) // crashes

      MyStrokedShape()
        .frame(width: 50, height: 200) // works

      MyStrokedAndFramedShape() // crashes
    }
  }
}
```

**Comment by j-f1** [comment]
Based on that, I wonder if `ModifiedContent` isn’t passing the environment to its child.

**Comment by mortenbekditlevsen** [comment]
Well, `frame()` applies a `ModifiedContent` and that is also used in some of the working examples.
Stroking is not applying `ModifiedContent`, but it does make a custom `Shape` into a `_ShapeView` with stroked contents.
Could it be in that 'transform' that things go wrong?

**Comment by mortenbekditlevsen** [comment]
Hi again @j-f1 ,
You're right. I am not certain what I was seing before, but what I can see now is that the `ModifiedContent` appears to not pass the environment to it's child.
In the `injectEnvironment` function, the `View` (i.e. in this case the `ModifiedContent`) has the environment set on all it's `EnvironmentReader` properties, but it's `Content` is not itself an `EnvironmentReader`, but rather another `View` which I am guessing ought to have `injectEnvironment` called directly.

But I don't quite get the relationship between the `MountedElement` construction and `ModifiedContent`.
In the case of the GTK renderer and the StaticHTML renderer, `ModifiedContent` appears both to be a `ParentView` and a `ViewDeferredToRenderer` when it's content is a `View`, so I don't know which elements actually are supposed to get mounted...

**Comment by MaxDesiatov** [comment]
@carson-katri I think you introduced `ModifiedContent` originally, maybe you could help?

**Comment by MaxDesiatov** [comment]
@ezraberch can we close this as completely resolved by #414?
